src/awkward/_v2/_connect/rdataframe/from_rdataframe.py

```diff
@@ -43,140 +48,214 @@
 assert done is True
 
 
-def from_rdataframe(data_frame, column):
-    def _wrap_as_record_array(array):
-        layout = array.layout if isinstance(array, ak._v2.highlevel.Array) else array
-        return ak._v2._util.wrap(
-            ak._v2.contents.RecordArray(
-                fields=[column],
-                contents=[layout],
-            ),
-            highlevel=True,
-        )
+def from_rdataframe(data_frame, columns):
+    def supported(form):
+        if form.purelist_depth == 1:
+            # special case for a list of strings form
+            return isinstance(
+                form, (ak._v2.forms.ListOffsetForm, ak._v2.forms.NumpyForm)
+            )
+        else:
+            return isinstance(form, ak._v2.forms.ListOffsetForm) and supported(
+                form.content
+            )
 
-    # Cast input node to base RNode type
-    data_frame_rnode = cppyy.gbl.ROOT.RDF.AsRNode(data_frame)
+    def form_dtype(form):
+        if form.purelist_depth == 1:
+            # special case for a list of strings form
+            return (
+                primitive_to_dtype(form.content.primitive)
+                if isinstance(form, ak._v2.forms.ListOffsetForm)
+                else primitive_to_dtype(form.primitive)
+            )
+        else:
+            return form_dtype(form.content)
```

Comment:
This function is similar to the "`supported`" function above, except that it wouldn't complain about any Form types that have a `content`, including all the list types (ListOffsetForm, ListForm, and RegularForm), but also anything at all with a `content`, such as IndexedOptionForm, IndexedForm, BitMaskedForm, ByteMaskedForm, UnmaskedForm.

Records and unions would simply fail with an AttributeError, since they don't have a `content`. Is this function protected somehow from these possibilities? That is, is there a proof that it will never be called with these other Forms as an input argument (`form`)?

Maybe the function should raise an error on these cases, but it likely needs to be something informative for users, not "`AttributeError: UnionForm does not have an attribute 'content'`".

Reply:
I can answer `__cpp_name__`, it is a `cppyy` built-in: https://cppyy.readthedocs.io/en/latest/misc.html#:~:text=__cpp_name__%3A%20a%20string%20that%20every%20C%2B%2B%20bound%20class%20carries%20and%20contains%20the%20actual%20C%2B%2B%20name%20(as%20opposed%20to%20__name__%20which%20has%20the%20Python%20name).%20This%20can%20be%20useful%20for%20template%20instantiations%2C%20documentation%2C%20etc.

Reply:
done